agent/src/test/java/com/teamscale/jacoco/agent/options/AgentOptionsParserTest.java

```diff
@@ -3,16 +3,40 @@
 import com.teamscale.jacoco.agent.util.TestUtils;
 import com.teamscale.report.util.CommandLineLogger;
 import okhttp3.HttpUrl;
-import org.assertj.core.api.Assertions;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
 
 import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 /** Tests parsing of the agent's command line options. */
 public class AgentOptionsParserTest {
 
-	private final AgentOptionsParser parser = new AgentOptionsParser(new CommandLineLogger());
+	private final AgentOptionsParser parser = new AgentOptionsParser(new CommandLineLogger(), null, null);
+
+	@Test
+	public void environmentConfigOverridesCommandLineOptions(
+			@TempDir Path tempDir) throws IOException, AgentOptionParseException {
+		Path config = tempDir.resolve("config.properties");
+		Files.write(config, "debug=true".getBytes(StandardCharsets.UTF_8));
+		AgentOptionsParser parser = new AgentOptionsParser(new CommandLineLogger(), config.toString(), null);
+		AgentOptions options = parser.parse("debug=false");
```

Comment:
we could, but this would be a bigger change to how the option parsing works since that doesnt keep track of how often an option occurs yet. So I'll create a ticket for that instead. I'll document that env options override command line options to at least make it clear in the docs

Reply:
Created https://cqse.atlassian.net/browse/TS-36384 and moved documentation task to https://cqse.atlassian.net/browse/TS-36383